docs(examples): add necessary descriptions for demo examples (#107)
AlaskaAirlines/auro-select: demo/demo.md

### Example with auro-icons in options

> Displays a baseline `<auro-select>` element with an appropriate `<auro-icon>` element next to each menu option.

Reviewer: ```suggestion
Displays an `<auro-select>` element with `<auro-icon>` elements in each option.
```

### Example with nested menus

> With the `<auro-menu>` supporting the placement of an `<auro-menu>` inside of another `<auro-menu>`, this example demonstrates what the selected option would look like at the root `<auro-menu>` or any of it's submenus.

Reviewer: ```suggestion
This example shows nesting `<auro-menu>` elements to create submenus.
```

### Example with no checkmark

> Applying the `noCheckmark` attribute in `<auro-select>` will pass down the attribute to `<auro-menu>`, preventing the check icon from being shown on the selected option. The left padding to reserve space for the checkmark is also removed.

Reviewer: ```suggestion
Applying the `noCheckmark` attribute will prevent the checkmark icon from being shown on the selected option. The left padding to reserve space for the checkmark will not be shown on options.
```
